types/vote_set.go

```diff
@@ -394,7 +403,7 @@ func (voteSet *VoteSet) HasTwoThirdsAny() bool {
 }
 
 func (voteSet *VoteSet) HasAll() bool {
-	return voteSet.sum == voteSet.valSet.TotalVotingPower()
+	return voteSet != nil && voteSet.sum == voteSet.valSet.TotalVotingPower()
```

Comment:
if we're calling these functions on a nil voteSet, something is probably very wrong. Perhaps we should panic? Maybe not HasAll, but I'd think for MakeCommit. Hard to always know ...

Reply:
You are right, I was just bullet proofing it since this VoteSet code is used in `consensus`.